server/index.js

```diff
@@ -145,6 +145,41 @@ app.post('/api/tasks/create', (req, res, next) => {
     .catch(err => next(err));
 });
 
+app.get('/api/tasks/view/:taskId', (req, res, next) => {
+
+  const taskId = Number(req.params.taskId);
+  if (!taskId) {
+    throw new ClientError(400, 'taskId must be a positive integer');
+  }
+
+  const sql = `
+    select "taskId",
+        "projectId",
+        "projectName",
+        "statusId",
+        "statusName",
+        "taskName",
+        "username",
+        "dateCreated",
+        "deadline",
+        "description"
+    from "tasks"
+    join "project" using ("projectId")
+    join "statuses" using ("statusId")
+    join "users" using ("userId")
+    where "taskId" = ($1)
+  `;
+
+  const param = [taskId];
+
+  db.query(sql, param)
+    .then(result => {
+      res.status(200).json(result.rows);
```

Comment:
If a client asks for one thing, then you need to only send them one thing or throw a 404